static/js/common.js

```diff
@@ -540,32 +540,82 @@ function runPanelAggsQuery(data, panelId, chartType, dataType, panelIndex, query
     }
 }
 
-function runMetricsQuery(data, panelId, currentPanel, queryRes) {
+
+async function runMetricsQuery(data, panelId, currentPanel, queryRes) {
     $('body').css('cursor', 'progress');
-    if (queryRes) {
-        renderChartByChartType(data,queryRes,panelId,currentPanel)
-    } else {
-        $.ajax({
-            method: 'post',
-            url: 'metrics-explorer/api/v1/timeseries',
-            headers: {
-                'Content-Type': 'application/json; charset=utf-8',
-                'Accept': '*/*'
-            },
-            crossDomain: true,
-            dataType: 'json',
-            data: JSON.stringify(data)
-        })
-            .then((res) => {
-                renderChartByChartType(data,res,panelId,currentPanel)
-            })
-            .catch(function (xhr, err) {
-                if (xhr.status === 400) {
-                    panelProcessSearchError(xhr, panelId);
+    if (panelId == -1) { // for panel on the editPanelScreen page
+        $(".panelDisplay #panelLogResultsGrid").hide();
+        $(".panelDisplay #empty-response").empty();
+        $('.panelDisplay #corner-popup').hide();
+        $(".panelDisplay #empty-response").hide();
+        $('.panelDisplay .panEdit-panel').show();
+    } else { // for panels on the dashboard page
+        $(`#panel${panelId} #panelLogResultsGrid`).hide();
+        $(`#panel${panelId} #empty-response`).empty();
+        $(`#panel${panelId} #corner-popup`).hide();
+        $(`#panel${panelId} #empty-response`).hide();
+        $(`#panel${panelId} .panEdit-panel`).show();
+    }
+    let chartType = currentPanel.chartType;
+    if (chartType === 'number'){
+        let bigNumVal = null;
+        let dataType = currentPanel.dataType;
+        let rawTimeSeriesData;
+        for (const queryData of data.queriesData) {
+            rawTimeSeriesData = await fetchTimeSeriesData(queryData);
+            const parsedQueryObject = parsePromQL(queryData.queries[0].query);
+            await addQueryElementOnAlertEdit(queryData.queries[0].name, parsedQueryObject);
```

Comment:
this function should be renamed , since it is used both in alert and dashboards

Reply:
Updated it to  `addQueryElementForAlertAndPanel`